writeengine/splitter/we_filereadthread.cpp

```diff
@@ -451,9 +477,48 @@ void WEFileReadThread::openInFile()
 {
     try
     {
-        if (fSdh.getDebugLvl()) cout << "Input FileName: " << fInFileName << endl;
+        /*  If an S3 transfer
+            use ms3 lib to d/l data into mem
+            use boost::iostreams to wrap the mem in a stream interface
+            point infile's stream buffer to it.
+        */
+    
+        if (fSdh.getDebugLvl()) cout << "Input Filename: " << fInFileName << endl;
+
+        if (doS3Import)
+        {
+            size_t bufLen = 0;
+            if (buf)
+            {
+                s3Stream.reset();
+                arrSource.reset();
+                free(buf);
+                buf = NULL;
+            }
+            if (fSdh.getDebugLvl())
+                cout << "Downloading " << fInFileName << endl;
+            int err = ms3_get(s3Connection, s3Bucket.c_str(), fInFileName.c_str(),
+                &buf, &bufLen);
+            if (fSdh.getDebugLvl())
+                cout << "Download complete." << endl;
+            if (err)
+            {
+                ostringstream os;
+                if (ms3_server_error(s3Connection))
+                    os << "Download of '" << fInFileName << "' failed.  Error from the server: "
+                        << ms3_server_error(s3Connection);
+                else 
+                    os << "Download of '" << fInFileName << "' failed.  Got '" << ms3_error(err)
+                        << "'.";
+                throw runtime_error(os.str());
+            }
+            
+            arrSource.reset(new boost::iostreams::array_source((char *) buf, bufLen));
+            s3Stream.reset(new boost::iostreams::stream<boost::iostreams::array_source>(*arrSource));
+            fInFile.rdbuf(s3Stream->rdbuf());
+        }
```

Comment:
The s3 lib currently requires downloading a whole object into mem.  This can change once add'l functionality is added to the s3 lib.